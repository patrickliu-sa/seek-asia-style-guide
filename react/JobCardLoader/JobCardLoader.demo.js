import React from 'react';
import PropTypes from 'prop-types';
import JobCardLoader from './JobCardLoader';
import PageBlock from '../PageBlock/PageBlock';

const JobCardLoaderContainer = ({ component: DemoComponent, componentProps }) => {
  return (
    <PageBlock style={{ width: '100%' }}>
      <DemoComponent {...componentProps} />
    </PageBlock>
  );
};

JobCardLoaderContainer.propTypes = {
  component: PropTypes.element,
  componentProps: PropTypes.object
};

export default {
  route: '/jobCardLoader',
  title: 'Job Card Loader',
  component: JobCardLoader,
  container: JobCardLoaderContainer,
  initialProps: {},
  options: []
};
