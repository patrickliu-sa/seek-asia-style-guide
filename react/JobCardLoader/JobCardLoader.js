import React from 'react';
import Card from '../Card/Card';
import styles from './JobCardLoader.less';

const JobCardLoader = () => {
  return (
    <Card className={styles.root}>
      <div className={styles.loader}>
        <div className={styles.animatedBackground}>
          <div className={`${styles.backgroundMasker} ${styles.companyRight}`} />
          <div className={`${styles.backgroundMasker} ${styles.companyBottom}`} />
          <div className={`${styles.backgroundMasker} ${styles.headerRight}`} />
          <div className={`${styles.backgroundMasker} ${styles.headerBottom}`} />
          <div className={`${styles.backgroundMasker} ${styles.desc1Right}`} />
          <div className={`${styles.backgroundMasker} ${styles.desc1Bottom}`} />
          <div className={`${styles.backgroundMasker} ${styles.desc2Right}`} />
          <div className={`${styles.backgroundMasker} ${styles.desc2Bottom}`} />
          <div className={`${styles.backgroundMasker} ${styles.locationRight}`} />
          <div className={`${styles.backgroundMasker} ${styles.locationBottom}`} />
          <div className={`${styles.backgroundMasker} ${styles.salaryRight}`} />
          <div className={`${styles.backgroundMasker} ${styles.salaryBottom}`} />
          <div className={`${styles.backgroundMasker} ${styles.dateRight}`} />
        </div>
      </div>
    </Card>
  );
};

export default JobCardLoader;
