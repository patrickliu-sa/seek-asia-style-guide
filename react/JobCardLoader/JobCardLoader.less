@import (reference) "~seek-asia-style-guide/theme";
@job-card-loader-base-height: @grid-base * 3;
@job-card-loader-pic: 80px;

.root {
  width: 100%;
  margin-bottom: 0;
  border-style: solid;
  border-bottom-width: 1px;
  border-color: @saGrey4;
}

.loader {
  background: @saWhite;
  min-width: 100%;
  min-height: @grid-base * 48;
  padding: @grid-base * 4;
}

.animatedBackground {
  animation-duration: 1s;
  animation-fill-mode: forwards;
  animation-iteration-count: infinite;
  animation-name: placeHolderShimmer;
  animation-timing-function: linear;
  background: @sk-gray-lightest;
  background: linear-gradient(75deg, @sk-gray-light 8%, @sk-mid-gray-light 18%, @sk-gray-light 33%);
  background-size: 200% 100%;
  height: @grid-base * 48;
  position: relative;
}

@keyframes placeHolderShimmer {
  0% {
    background-position: 95% 0%;
  }
  100% {
    background-position: -95% 0%;
  }
}

.backgroundMasker {
  background: @saWhite;
  position: absolute;
}

.companyRight {
  top: 0;
  left: 30%;
  right: 0;
  height: @job-card-loader-base-height;
}

.companyBottom {
  top: @job-card-loader-base-height;
  left: 0;
  right: 0;
  height: @job-card-loader-base-height;
}

.headerRight {
  top: @job-card-loader-base-height * 2;
  left: 80%;
  right: 0;
  height: @job-card-loader-base-height;
}

.headerBottom {
  top: @job-card-loader-base-height * 3;
  left: 0;
  right: 0;
  height: @job-card-loader-base-height *2;
}

.desc1Right {
  top: @job-card-loader-base-height * 5;
  left: 65%;
  right: 0;
  height: @job-card-loader-base-height;
}

.desc1Bottom {
  top: @job-card-loader-base-height * 6;
  left: 0;
  right: 0;
  height: @job-card-loader-base-height;
}

.desc2Right {
  top: @job-card-loader-base-height * 7;
  left: 60%;
  right: 0;
  height: @job-card-loader-base-height;
}

.desc2Bottom {
  top: @job-card-loader-base-height * 8;
  left: 0;
  right: 0;
  height: @job-card-loader-base-height * 2;
}

.locationRight {
  top: @job-card-loader-base-height * 10;
  left: 45%;
  right: 0;
  height: @job-card-loader-base-height;
}

.locationBottom {
  top: @job-card-loader-base-height * 11;
  left: 0;
  right: @job-card-loader-pic;
  height: @job-card-loader-base-height;
}

.salaryRight {
  top: @job-card-loader-base-height * 12;
  left: 50%;
  right: @job-card-loader-pic;
  height: @job-card-loader-base-height;
}

.salaryBottom {
  top: @job-card-loader-base-height * 13;
  left: 0;
  right: @job-card-loader-pic;
  height: @job-card-loader-base-height * 2;
}

.dateRight {
  top: @job-card-loader-base-height * 15;
  left: @grid-base * 10;
  right: @job-card-loader-pic;
  height: @job-card-loader-base-height;
}
