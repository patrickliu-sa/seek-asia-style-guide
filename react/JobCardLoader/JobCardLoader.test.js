import React from 'react';
import { shallow } from 'enzyme';

import JobCardLoader from './JobCardLoader';

describe('JobCardLoader', () => {
  it('should render with defaults', () => {
    expect(shallow(<JobCardLoader />)).toMatchSnapshot();
  });
});
