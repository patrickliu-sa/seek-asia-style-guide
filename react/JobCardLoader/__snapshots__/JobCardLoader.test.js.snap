// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCardLoader should render with defaults 1`] = `
<Card
  className="root"
>
  <div
    className="loader"
  >
    <div
      className="animatedBackground"
    >
      <div
        className="backgroundMasker companyRight"
      />
      <div
        className="backgroundMasker companyBottom"
      />
      <div
        className="backgroundMasker headerRight"
      />
      <div
        className="backgroundMasker headerBottom"
      />
      <div
        className="backgroundMasker desc1Right"
      />
      <div
        className="backgroundMasker desc1Bottom"
      />
      <div
        className="backgroundMasker desc2Right"
      />
      <div
        className="backgroundMasker desc2Bottom"
      />
      <div
        className="backgroundMasker locationRight"
      />
      <div
        className="backgroundMasker locationBottom"
      />
      <div
        className="backgroundMasker salaryRight"
      />
      <div
        className="backgroundMasker salaryBottom"
      />
      <div
        className="backgroundMasker dateRight"
      />
    </div>
  </div>
</Card>
`;
